fedimint-api/Cargo.toml

```diff
@@ -25,6 +25,7 @@ rand07 = { version = "0.7", package = "rand" }
 secp256k1-zkp = { version = "0.6.0", features = [ "use-serde", "bitcoin_hashes", "global-context" ] }
 serde = { version = "1.0.145", features = [ "derive" ] }
 serde_json = "1.0.85"
+strum = { version = "0.24.1", features = ["derive"] }
```

Comment:
If I'm seeing it right, this could be a dev-depenendecny? (sorry if I missed something)

Reply:
I think you are correct where this draft PR (in its current state) is concerned. But I would like to be able to iterate over enum fields in non-dev where dbdump is concerned, see issue https://github.com/fedimint/fedimint/issues/331.

Reply:
Do we really need this dep? If it's just for iterating over enum variants we should probably just use a macro that implements this since `strum` seems to do a lot more.

Reply:
Copying the `strum::EnumIter` macro from `strum` would mean adding at least 160 lines of code and a new dependency on `syn`. I haven't investigated whether it might be possible to do the same thing with less code.

Personally I would lean towards keeping things simple and using `strum`. I think that the tradeoff of an additional dependency is worth it.

But let me know what the maintainers want and I'll do what's necessary.